core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/convertRowCellImpl.kt

```diff
@@ -0,0 +1,27 @@
+package org.jetbrains.kotlinx.dataframe.impl.api
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.RowValueExpression
+import org.jetbrains.kotlinx.dataframe.api.Convert
+import org.jetbrains.kotlinx.dataframe.api.Infer
+import org.jetbrains.kotlinx.dataframe.api.asColumn
+import org.jetbrains.kotlinx.dataframe.api.name
+import org.jetbrains.kotlinx.dataframe.exceptions.ColumnTypeMismatchesColumnValuesException
+import org.jetbrains.kotlinx.dataframe.impl.columns.newColumn
+import kotlin.reflect.KType
+
+@PublishedApi
```

Comment:
Could you maybe add a comment for this function (either as `//` or `/** @comment */` why this function needs to be in this file and not in `convert.kt`? I'm afraid we won't remember anymore why we moved it in a few years.

Reply:
Also, if we follow our naming convention, the file should probably be named `withRowCell.kt`. Maybe if we do that, we're less inclined to move it in the future ;P

Reply:
Those help, but I don't consider them a main source-of-truth, as refactorings etc. can sometimes alter them